vendor/forms.py

```diff
@@ -387,6 +391,62 @@ def clean(self):
 
         return cleaned_data
 
+
+class SiteSelectForm(forms.Form):
+    site = forms.ModelChoiceField(queryset=Site.objects.all())
+
+class OfferSiteSelectForm(SiteSelectForm):
+    offer = forms.ModelChoiceField(queryset=None, required=False)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if 'site' in self.initial or 'site' in self.data:
+            site = self.initial.get('site') if self.initial.get('site', None) else self.data['site']
+            self.fields['offer'].queryset = Offer.objects.filter(site=site)
+        else:
+            self.fields['offer'].widget = forms.HiddenInput()
+
+
+class SubscriptionForm(forms.Form):
+    site = forms.ModelChoiceField(queryset=Site.objects.all())
+    profile = forms.ModelChoiceField(queryset=None)
+    subscription_id = forms.CharField(max_length=80)
+    status = forms.ChoiceField(choices=SubscriptionStatus.choices)
+        
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if not ('site' in self.initial or 'site' in self.data):
+            raise KeyError("site needs to be inluded in either initial or data attributes")
+
+        site = self.initial.get('site') if self.initial.get('site', None) else self.data['site']
+            
+        self.fields['site'].widget = forms.HiddenInput()
```

Comment:
just wondering why we're using the site as a form field when it's always hidden from the user... might be better as a form kwarg if there's no way to select it in the form? not a blocker